api/rosetta/retriever.go

```diff
@@ -28,4 +30,5 @@ type Retriever interface {
 	Transaction(rosBlockID identifier.Block, rosTxID identifier.Transaction) (*object.Transaction, error)
 	Balances(rosBlockID identifier.Block, rosAccountID identifier.Account, rosCurrencies []identifier.Currency) (identifier.Block, []object.Amount, error)
 	SequenceNumber(rosAccountID identifier.Account, keyIndex int) (uint64, error)
+	Key(rosAccountID identifier.Account, keyIndex int) (*flow.AccountPublicKey, error)
```

Comment:
I would completely remove the `SequenceNumber` function from this interface and just use the `Key` (perhaps rename to `PublicKey`?).

I've left it like this in this PR because of the difference between `invalid proposal key` and `invalid authorizer key` errors (even though they are basically the same case).

Perhaps it makes sense to just have an error type of `failure.InvalidKey` and from there, depending on the context, return `invalidProposalKey()` or `invalidAuthorizerKey()`. Both error functions would then just refere to `failure.InvalidKey`, instead of the special cases.

What do you think?

Reply:
I'm on board with the idea behind the comment, though I'm not sure the details are spot on yet.